be/src/connector/deletion_vector/deletion_vector.cpp

```diff
@@ -0,0 +1,170 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "deletion_vector.h"
+
+#include <roaring/roaring64.h>
+
+#include "util/base85.h"
+#include "util/uuid_generator.h"
+
+namespace starrocks {
+
+Status starrocks::DeletionVector::fill_row_indexes(std::set<int64_t>* need_skip_rowids) {
+    if (_deletion_vector_descriptor->__isset.cardinality && _deletion_vector_descriptor->__isset.cardinality == 0) {
+        return Status::OK();
+    } else if (is_inline()) {
+        return deserialized_inline_dv(_deletion_vector_descriptor->pathOrInlineDv, need_skip_rowids);
+    } else {
+        std::shared_ptr<io::SharedBufferedInputStream> shared_buffered_input_stream = nullptr;
+        std::shared_ptr<io::CacheInputStream> cache_input_stream = nullptr;
+        HdfsScanStats app_scan_stats;
+        HdfsScanStats fs_scan_stats;
+
+        ASSIGN_OR_RETURN(auto path, get_absolute_path(_params.table_location));
+        int64_t offset = _deletion_vector_descriptor->offset;
+        int64_t length = _deletion_vector_descriptor->sizeInBytes;
+
+        ASSIGN_OR_RETURN(auto dv_file, open_random_access_file(path, fs_scan_stats, app_scan_stats,
+                                                               shared_buffered_input_stream, cache_input_stream));
+        // Check the dv size
+        uint32_t size_from_deletion_vector_file;
+        RETURN_IF_ERROR(dv_file->read_at_fully(offset, &size_from_deletion_vector_file, DV_SIZE_LENGTH));
+        offset += DV_SIZE_LENGTH;
+        // the size_from_deletion_vector_file is big endian byte order
+        if (LittleEndian::IsLittleEndian()) {
+            size_from_deletion_vector_file = BigEndian::ToHost32(size_from_deletion_vector_file);
+        }
+
+        if (size_from_deletion_vector_file != length) {
+            std::stringstream ss;
+            ss << "DV size mismatch, expected : " << length << " , actual : " << size_from_deletion_vector_file;
+            return Status::RuntimeError(ss.str());
+        }
+
+        // Check the correctness of magic number
+        uint32_t magic_number_from_deletion_vector_file;
+        RETURN_IF_ERROR(dv_file->read_at_fully(offset, &magic_number_from_deletion_vector_file, MAGIC_NUMBER_LENGTH));
+        // magic_number_from_deletion_vector_file is little endian byte order
+        if (!LittleEndian::IsLittleEndian()) {
+            magic_number_from_deletion_vector_file = LittleEndian::ToHost32(magic_number_from_deletion_vector_file);
+        }
+        offset += MAGIC_NUMBER_LENGTH;
+
+        int64_t serialized_bitmap_length = length - MAGIC_NUMBER_LENGTH;
+        std::unique_ptr<char[]> deletion_vector(new char[serialized_bitmap_length]);
```

Comment:
you can just use std::vector<char>.  

unique_ptr<char[]> is little weird.

Reply:
OK. I will modify this next PR